packages/worker/src/runtime.ts

```diff
@@ -265,31 +265,42 @@ export class Runtime {
     };
   }
 
-  protected async initLogPolling(logger: CoreLogger): Promise<void> {
-    this.shouldPollForLogs.next(true);
-
+  protected async initLogPolling(logger: BufferedLogger): Promise<void> {
     if (!this.isForwardingLogs()) {
       return;
     }
+
+    const stopPollingForLogs = firstValueFrom(this.stopPollingForLogs);
+
     const poll = promisify(pollLogs);
+    const doPoll = async () => {
+      const logs = await poll(this.native);
+      for (const log of logs) {
+        const meta: Record<string | symbol, unknown> = {
+          [LogTimestamp]: timeOfDayToBigint(log.timestamp),
+        };
+        if (log.fields) {
+          for (const [key, value] of Object.entries(log.fields)) {
+            meta[key] = JSON.parse(value);
+          }
```

Comment:
This is the only bit that is maybe mildly worrisome, doing quite a bit more parsing now. That's really got to happen though without heading down the flatbuffers road or something, so it's not like there's an alternative.

Otherwise this change looks totally fine to me pending what we decide about key names

Reply:
Agree, I'm a bit worried about the perf implications here.

Reply:
I do share that worry, though I'm not sure if this is actually a thing. TBH that's pretty much _the_ thing I really wanted opinions about.

The perf impact pretty much comes down to the number and complexity of metadata that are produced by Core. Converting a few integers to string then back to numbers is definitely not optimal, yet not that bad. Same for a few short strings composed of almost ascii-7 characters. Converting large/complex JSON objects would definitely be something else. But at this point, I'm not sure what I should expect to find out in these metadata.

> it's not like there's an alternative

Well, sure there is. I ended up going for JSON just for simplicity, but I was initially thinking of recursively creating the JS object tree directly in Rust code ([here](https://github.com/temporalio/sdk-typescript/pull/1225/files#diff-198eab910c3297070c03356ffb4f223a429afb656d8bf06ee59132bea522223bR210-R215)). I still consider that as an option. Should not be a difficult thing.

Reply:
Yeah, if we have the rust types and it’s all primitives, I would do the conversion at the bridge

Reply:
Ah, yeah that makes perfect sense. I think I'd go for that. Though _honestly_, it is not immediately obvious to me it'd actually be faster. JSON parsers are one of the most optimized things on the planet and I wouldn't be completely surprised to find that it actually ends up being faster than constructing the JS objects in the bridge and passing them back, since I'm not entirely sure what that entails.

We'd need to actually bench it. However, I don't think the actual impact is going to be super meaningful one way or the other. Not sure it's worth spending the time on that.